pkg/image/registry/imagestreamtag/rest.go

```diff
@@ -111,20 +116,30 @@ func (r *REST) Delete(ctx kapi.Context, id string) (runtime.Object, error) {
 		return nil, err
 	}
 
-	if stream.Spec.Tags == nil {
-		return nil, errors.NewNotFound("imageStreamTag", tag)
+	notFound := true
+
+	// Try to delete the status tag
+	if stream.Status.Tags != nil {
+		if _, ok := stream.Status.Tags[tag]; ok {
```

Comment:
The internal if will return you false when the tag is not found, even when map is not initiated. So the outer check is not needed, this will simplify this code a bit. The same applies in the 2nd if.